docs/architecture/index.md

```diff
@@ -0,0 +1,9 @@
+# Architecture
+
+This living document serves as a high-level documentation to explain the conceptual choices and tool choices in the Gutenberg repository.
+
+- [Modularity and WordPress Packages](docs/architecture/modularity.md).
+- [Block Editor Performance](docs/architecture/performance.md).
+- What are the decision decisions behind the Data Module?
+- Why puppeteer is the tool of choice for End 2 End tests?
+- What's the difference between the different editor packages? and what's the purpose of each package?
```

Comment:
I'm hoping that by starting this, other folks can chime-in and document other decisions made on the repo.